streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -483,6 +552,79 @@ public static void waitUntilMetadataIsPropagated(final List<KafkaServer> servers
 
     }
 
+    public static <K, V> void verifyOutput(final String broker,
+                                           final String topic,
+                                           final Deserializer<K> keyDeserializer,
+                                           final Deserializer<V> valueDeserializer,
+                                           final List<KeyValueTimestamp<String, Long>> expected) {
+        final List<ConsumerRecord<String, Long>> results;
+        try {
+            final Properties properties = mkProperties(
+                mkMap(
+                    mkEntry(ConsumerConfig.GROUP_ID_CONFIG, "test-group"),
+                    mkEntry(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, broker),
+                    mkEntry(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer.getClass().getName()),
+                    mkEntry(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer.getClass().getName())
+                )
+            );
+            results = IntegrationTestUtils.waitUntilMinRecordsReceived(properties, topic, expected.size());
+        } catch (final InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (results.size() != expected.size()) {
+            throw new AssertionError(printRecords(results) + " != " + expected);
+        }
+        final Iterator<KeyValueTimestamp<String, Long>> expectedIterator = expected.iterator();
+        for (final ConsumerRecord<String, Long> result : results) {
+            final KeyValueTimestamp<String, Long> expected1 = expectedIterator.next();
+            try {
+                compareKeyValueTimestamp(result, expected1.key(), expected1.value(), expected1.timestamp());
+            } catch (final AssertionError e) {
+                throw new AssertionError(printRecords(results) + " != " + expected, e);
+            }
+        }
+    }
+
+    private static <K, V> void compareKeyValueTimestamp(final ConsumerRecord<K, V> record,
+                                                        final K expectedKey,
+                                                        final V expectedValue,
+                                                        final long expectedTimestamp) {
+        Objects.requireNonNull(record);
+        final K recordKey = record.key();
+        final V recordValue = record.value();
+        final long recordTimestamp = record.timestamp();
+        final AssertionError error = new AssertionError("Expected <" + expectedKey + ", " + expectedValue + "> with timestamp=" + expectedTimestamp +
+                                                            " but was <" + recordKey + ", " + recordValue + "> with timestamp=" + recordTimestamp);
```

Comment:
this verification block duplicates code in `OutputVerifier`, refactoring to some common verification method is out of scope for this PR, but I wanted to get my thoughts out on this as it seems some of our testing code can be consolidated some.

\cc @guozhangwang @mjsax

Reply:
I hear you, but in this case, I would resist consolidation.

OutputVerifier is a public test utility for use with the TopologyTestDriver. Its behavior needs to be stable over time.

IntegrationTestUtils is a private utility for use in our integration tests (i.e., not TopologyTestDriver). We should maintain the flexibility to change these utilities completely at will.

So it seems they have little in common except that they need to print a message when an assertion is false during testing. The message is for human consumption, so its format doesn't need to be any particular way, just as long as it's clear and informative.

IMHO, these two components are far apart in usage and modular structure, and there doesn't seem to be a compelling benefit from coupling them.

Reply:
I don't care too much personally. However, I don't think it's a issue to use `OutputVerifier` in unit test (if we can pull it in, with dependency issues?) as long as it does what we need. Of course, we would not change `OutputVerifier` is it does not meet the unit test requirements but would move off it for this case.

Reply:
Also, note that this is a `ConsumerRecord`, and `OutputVerifier` is for a `ProducerRecord`.